Destination/RotatableStreamDestination.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Zenstruck\BackupBundle\Destination;
+
+use Psr\Log\LoggerInterface;
+use Zenstruck\BackupBundle\Rotator\Rotator;
+
+class RotatableStreamDestination extends StreamDestination
+{
+    /**
+     * @var null|Rotator
+     */
+    private $preRotator;
+
+    /**
+     * @var null|Rotator
+     */
+    private $postRotator;
+
+    /**
+     * A constructor
+     *
+     * @param string $directory Path to backup directory.
+     * @param Rotator|null $preRotator Rotator to execute prior to push.
+     * @param Rotator|null $postRotator Rotator to execute after push.
+     */
+    public function __construct($directory, Rotator $preRotator = null, Rotator $postRotator = null)
+    {
+        parent::__construct($directory);
+        $this->preRotator = $preRotator;
+        $this->postRotator = $postRotator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function push($filename, LoggerInterface $logger)
```

Comment:
Multiple inheritance is not possible, and you want to be 5.3 compatible, so I can not use Traits. So this is little bit redundant code - for any new implementation of destination, same 'preRotator' / 'postRotator' logic would have to be implemented. For now, we are fine... we will see in the future how to handle this.... I was thinking about Gaufrete as Destination driver implementation as well...